jscomp/bsb/bsb_main.ml

```diff
@@ -239,45 +239,40 @@ let print_string_args (args : string array) =
 (* Note that [keepdepfile] only makes sense when combined with [deps] for optimization
    It has to be the last command of [bsb]
 *)
-let exec_command_install_then_exit  command =
-  Format.fprintf Format.std_formatter "@{<info>CMD:@} %s@." command;
-  print_endline command ;
-  exit (Sys.command command ) 
-
-(* Execute the underlying ninja build call, then exit (as opposed to keep watching) *)
-let ninja_command_exit  vendor_ninja ninja_args  =
+let ninja_command_exit  ?warnings:(warnings=[]) vendor_ninja ninja_args  =
   let ninja_args_len = Array.length ninja_args in
+  let prevEnv = Unix.environment () in
+  let newEnv = if warnings <> [] then
+    let prevEnvLength = Array.length prevEnv in
+    Array.init (prevEnvLength + 1)
+      (fun i -> if i < prevEnvLength then 
+        Array.unsafe_get prevEnv i
+      else
+        (List.fold_left (fun acc w -> acc ^ w) "OCAMLPARAM=w=" warnings) ^ ",_")
```

Comment:
about windows: https://superuser.com/questions/223104/setting-and-using-variable-within-same-command-line-in-windows-cmd-exe